api/internal/builtins/SortOrderTransformer.go

```diff
@@ -87,19 +87,10 @@ func (p *SortOrderTransformerPlugin) Transform(m resmap.ResMap) (err error) {
 // applyOrdering takes resources (given in ResMap) and a desired ordering given
 // as a sequence of ResIds, and updates the ResMap's resources to match the
 // ordering.
-func applyOrdering(m resmap.ResMap, ordering []resid.ResId) error {
-	var err error
-	resources := make([]*resource.Resource, m.Size())
-	// Clear and refill with the correct order
-	for i, id := range ordering {
-		resources[i], err = m.GetByCurrentId(id)
-		if err != nil {
-			return errors.WrapPrefixf(err, "expected match for sorting")
-		}
-	}
+func applyOrdering(m resmap.ResMap, resources []*resource.Resource) error {
 	m.Clear()
 	for _, r := range resources {
-		err = m.Append(r)
+		err := m.Append(r)
 		if err != nil {
 			return errors.WrapPrefixf(err, "SortOrderTransformer: Failed to append to resources")
 		}
```

Comment:
I think there's no longer a need for a `applyOrdering` function.
The resources are already ordered so the function doesn't do much anymore.
Might as do that in the `Transform` function or rename this to `replaceResMapResources` or similar.

Reply:
I will do a follow-up with another performance improvement in this file, so I hope we could merge this first and then do more cleanups there or later. I hope that's OK